Hierarchical3DRegistration/Hierarchical3DRegistrationLib/TreeNode.py

```diff
@@ -73,15 +73,80 @@ def __init__(
 
     def _generateRoiFromModel(self) -> slicer.vtkMRMLMarkupsROINode:
         """Creates a region of interest node from this TreeNode's model."""
+        import SegmentStatistics
+        import vtk
+
+        segNode = slicer.vtkMRMLSegmentationNode()
+        slicer.mrmlScene.AddNode(segNode)
+        segNode.CreateDefaultDisplayNodes()
+        segNode.SetReferenceImageGeometryParameterFromVolumeNode(self.sourceVolume)
+
+        slicer.modules.segmentations.logic().ImportModelToSegmentationNode(self.model, segNode)
+        # Compute centroids
+
+        segStatLogic = SegmentStatistics.SegmentStatisticsLogic()
+        segStatLogic.getParameterNode().SetParameter("Segmentation", segNode.GetID())
+        segStatLogic.getParameterNode().SetParameter("LabelmapSegmentStatisticsPlugin.centroid_ras.enabled", str(True))
+        segStatLogic.getParameterNode().SetParameter(
+            "LabelmapSegmentStatisticsPlugin.principal_axis_x.enabled", str(True)
+        )
+        segStatLogic.getParameterNode().SetParameter(
+            "LabelmapSegmentStatisticsPlugin.principal_axis_y.enabled", str(True)
+        )
+        segStatLogic.getParameterNode().SetParameter(
+            "LabelmapSegmentStatisticsPlugin.principal_axis_z.enabled", str(True)
+        )
```

Comment:
Optional, but we could do something like this here to avoid the line wraps
```suggestion
        segStatParams = segStatLogic.getParameterNode()
        segStatParams.SetParameter("Segmentation", segNode.GetID())
        segStatParams.SetParameter("LabelmapSegmentStatisticsPlugin.centroid_ras.enabled", str(True))
        segStatParams.SetParameter("LabelmapSegmentStatisticsPlugin.principal_axis_x.enabled", str(True))
        segStatParams.SetParameter("LabelmapSegmentStatisticsPlugin.principal_axis_y.enabled", str(True))
        segStatParams.SetParameter("LabelmapSegmentStatisticsPlugin.principal_axis_z.enabled", str(True))
```